pkg/actuators/machine/reconciler.go

```diff
@@ -40,38 +50,200 @@ func newReconciler(scope *machineScope) *Reconciler {
 	}
 }
 
+// Create creates an instance via machine cr which is handled by cluster-api
 func (r *Reconciler) create() error {
-	klog.Infof("%s: creating machine", r.machine.Name)
 
-	if err := validateMachine(*r.machine, *r.providerSpec); err != nil {
+	if err := validateMachine(*r.machine); err != nil {
 		return machinecontroller.InvalidMachineConfiguration("failed validating machine provider spec: %v", err)
 	}
 
-	return r.requeueIfInstancePending()
+	userData, err := r.getUserData()
+	if err != nil {
+		return fmt.Errorf("failed to get user data: %w", err)
+	}
+
+	// Create an instance
+	_, err = r.ibmClient.InstanceCreate(r.machine.Name, r.providerSpec, userData)
+
+	if err != nil {
+		klog.Errorf("%s: error occured while creating machine: %w", r.machine.Name, err)
+		metrics.RegisterFailedInstanceCreate(&metrics.MachineLabels{
+			Name:      r.machine.Name,
+			Namespace: r.machine.Namespace,
+			Reason:    err.Error(),
+		})
+
+		if reconcileMachineWithCloudStateErr := r.reconcileMachineWithCloudState(&ibmcloudproviderv1.IBMCloudMachineProviderCondition{
+			Type:    ibmcloudproviderv1.MachineCreated,
+			Status:  apicorev1.ConditionFalse,
+			Reason:  ibmcloudproviderv1.MachineCreationFailed,
+			Message: err.Error(),
+		}); reconcileMachineWithCloudStateErr != nil {
+			klog.Errorf("failed to reconcile machine condtion with cloud state: %v", reconcileMachineWithCloudStateErr)
+		}
+		return fmt.Errorf("failed to create instance via ibm vpc client: %w", err)
+	}
+
+	// Update Machine Spec and status with instance info
+	return r.reconcileMachineWithCloudState(nil)
 }
 
+// update gets instance details and reconciles the machine resource with its state
 func (r *Reconciler) update() error {
+	if err := validateMachine(*r.machine); err != nil {
+		return machinecontroller.InvalidMachineConfiguration("failed validating machine provider spec: %v", err)
+	}
+
+	// Update cloud state
+	return r.reconcileMachineWithCloudState(nil)
+}
+
+func validateMachine(machine machinev1.Machine) error {
+	if machine.Labels[machinev1.MachineClusterIDLabel] == "" {
+		return machinecontroller.InvalidMachineConfiguration("machine is missing %q label", machinev1.MachineClusterIDLabel)
+	}
+
 	return nil
 }
 
 // Returns true if machine exists.
 func (r *Reconciler) exists() (bool, error) {
-	return true, nil
+	// check if instance exist
+	exist, err := r.ibmClient.InstanceExistsByName(r.machine.GetName(), r.providerSpec)
+
```

Comment:
Might be good to log an error here if err != nil

Reply:
The controller should log if `err != nil`

Catching and returning the err in `InstanceExistsByName()` in `client.go`

..If we log here then we will be logging it twice.